### ISSUE TITLE
Do the tsfresh packaged provide batch processing over a distributed architecture or not?

### ISSUE THREAD

**Issue by ricardovvargas** [blue-yonder/tsfresh#449]
Hi. I am confused because the [introduction section](https://tsfresh.readthedocs.io/en/latest/text/introduction.html) of the tsfresh documentation says that "tsfresh is not suitable for batch processing over a distributed architecture, where different time series are fragmented over different computational units" but [this page](https://tsfresh.readthedocs.io/en/latest/text/tsfresh_on_a_cluster.html) says the following: "Indeed, it is possible to extract features with tsfresh in a distributed fashion."

Did I miss something or is this just a case of outdated documentation?

**Comment by MaxBenChrist** [comment]
Both statements are correct and indeed tsfresh will run on cluster. (I ran tsfresh on 100 of machines for weeks on huge time series data sets).

Now, the part 

> "tsfresh is not suitable for batch processing over a distributed architecture, where different time series are fragmented over different computational units"

tries to explain that all time series need to be on one machine, the dask master. So, if your time series are fragmented over the cluster, you will need to consolidate it on one machine before starting the feature extraction. Its a technical detail and the statement is probably a little bit confusing, will see if I can formulate it more clearly.

**Comment by ricardovvargas** [comment]
Humm... Now it is clear. Thanks for this explanation. My mother language is brazilian portuguese and I am not an english grammar specialist but as far as I know the term "where" explain something about the last sentence while the term "when" do a restriction. So I believe that the following minor change would make this part of the documentation much more clear: "tsfresh is not suitable for batch processing over a distributed architecture when different time series are fragmented over different computational units". Anyway, this is just a suggestion. Tsfresh is a very nice package. Congratulations.

**Comment by ricardovvargas** [comment]
Yes, sure. I really intend to use this great package at work (Petrobras S.A) and in my doctoral project and, as a counterpart, collaborate as much as I can. The problem is that I am not so familiar with working with GitHub yet : - ))

After clicking the "New pull request", what am I supposed to do? Do you suggest any tutorial that I can follow to avoid messing up with the project or this repository?
